mobile/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -0,0 +1,245 @@
+
+      throw "ERROR: Source table is missing column(s) which exist in target table.";
+
+  } else {
+
+      var columnAdditions = sourceColumns.filter(notIncludedIn(targetColumns));
+
+      if (sourceColumns.some(notIncludedIn(targetColumns)) === true) {
+          if (!AUTOMIGRATE) {
+              throw "ERROR: Target table is missing column(s),but automigrate is not enabled.";
+          } else {
+              // enforce order
+              for (var i = 0; i < targetColumns.length; i++) {
+                  if (targetColumns[i] !== sourceColumns[i]) {
+                      throw "ERROR: Can only migrate extra columns of the end of source"
+                  }
+              }
+              add_columns_to(TARGET_SCHEMA, TARGET_TABLE, columnAdditions.join(','));
+              return "ok.Columns added."
+          }
+      } else {
+          return "ok. Columns match."
+      }
+  }
+
+  // == Helpers ==
+
+  function list_cols_with_type(sch,tbl,delimiter) {
+      var stmt = `
+          SELECT
+              LISTAGG(
+                  CASE
+                    WHEN isc.data_type='TEXT'
+                      THEN CONCAT(isc.column_name, ' VARCHAR(',isc.character_maximum_length, ')')
+                    WHEN isc.data_type='NUMBER'
+                      THEN CONCAT(isc.column_name, ' NUMBER(', isc.numeric_precision, ',',isc.numeric_scale, ')')
+                    ELSE
+                      CONCAT(isc.column_name, ' ', isc.data_type)
+                  END, '` + delimiter + `')
+              WITHIN GROUP (order by isc.ordinal_position)
+          FROM information_schema.columns AS isc
+          WHERE table_schema = UPPER('` + sch + `')
+            AND table_name = UPPER('` + tbl + `');`;
+
+      var res = snowflake.createStatement({sqlText: stmt}).execute();
+      res.next();
+      result = res.getColumnValue(1);
+
+      return result;
+  }
+
+  function notIncludedIn(arr) {
+      return function(elt) {
+        return ! arr.includes(elt);
+      };
+  }
+
+  function add_columns_to(sch, tbl, cols) {
+      var alter_stmt = `ALTER TABLE ` + sch + `.` + tbl + ` ADD COLUMN ` + cols;
+      snowflake.createStatement({sqlText: alter_stmt}).execute();
+
+      return "ok. Columns added.";
+  }
+
+  $$
+;
+
+
+/*
+  COMMIT_TABLE
+  Inputs:
+  SOURCE_SCHEMA:         the schema of the source table
+  SOURCE_TABLE:          the source table to select data from
+  TARGET_SCHEMA:         the schema of the target table
+  TARGET_TABLE:          the target table to delete/insert into
+  JOIN_KEY:              the join key between the source and target tables
+  PARTITION_KEY:         the partition of the target table
+  AUTOMIGRATE:           whether to automigrate staged table
+  To drop:
+  DROP PROCEDURE {{.output_schema}}.commit_table(VARCHAR,VARCHAR,VARCHAR,VARCHAR,VARCHAR,VARCHAR,BOOLEAN);
+*/
+CREATE OR REPLACE PROCEDURE {{.output_schema}}.commit_table(SOURCE_SCHEMA VARCHAR,
```

Comment:
We should indeed rename this function - since we have two distinct functions with the same arguments of different types, this isn't a safe way to use overloading.

Snowflake will cast the argument if it's the wrong type, which will cause unexpected behaviour: https://docs.snowflake.com/en/sql-reference/stored-procedures-usage.html#overloading-stored-procedure-names

Reply:
Ok it is actually the `column_check` procedure that was changed between the web and mobile. I am removing this procedure entirely from this version of the model so I don't think there will be any naming clashes now

Reply:
@bill-warner `commit_table` is still best off need being renamed as it is still divergent from the function of the same name in the web models.